sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -329,6 +329,38 @@ struct _pi_device : _pi_object {
   ZeStruct<ze_device_compute_properties_t> ZeDeviceComputeProperties;
 };
 
+// Structure describing the specific use of a command-list in a queue.
+// This is because command-lists are re-used across multiple queues
+// in the same context.
+typedef struct {
+  // The Level-Zero fence that will be signalled at completion.
+  ze_fence_handle_t ZeFence{nullptr};
+  // Record if the fence is in use.
+  // This is needed to avoid leak of the tracked command-list if the fence
+  // was not yet signaled at the time all events in that list were already
+  // completed (we are polling the fence at events completion). The fence
+  // may be still "in-use" due to sporadic delay in HW.
+  bool InUse{false};
+  // Record if the command list is a "copy" command list.
+  bool IsCopy{false};
+
+  // Keeps events created by commands submitted into this command-list.
+  // TODO: use this for explicit wait/cleanup of events at command-list
+  // completion.
+  // TODO: use this for optimizing events in the same command-list, e.g.
+  // only have last one visible to the host.
+  std::vector<pi_event> EventList{};
+  size_t size() const { return EventList.size(); }
+  void append(pi_event Event) { EventList.push_back(Event); }
+
+} pi_command_list_info_t;
```

Comment:
How about creating a _pi_command_list struct? It will have all the fields that you have in pi_command_list_info_t and also have an additional field to hold the ZeCommandList ptr. Then pi_queue can have an unordered map from ZeCommandList to _pi_command_list.
This makes the code more structured as _pi_command_list will have a structure similar to _pi_queue, _pi_device, and _pi_event

Reply:
I did initially do that, but than realized that there is data duplication this way. The entry in the map already keeps L0 command-list handle (as a key), and repeating it in the value is a waste of space with no apparent advantage in anything else.

Reply:
Please let me know what you think about this proposal. I have other comments based on whether we want to adopt this proposal or not. Thanks.

Reply:
see just above

Reply:
I agree. How about we keep the naming consistent with other structs..Instead of calling it _pi_command_list_info, we can just call it _pi_command_list?
Also, I see all the other _pi_* structs derived from _pi_object. May be _pi_command_list can also derive out of it?

Reply:
I exactly wanted to NOT follow other PI objects because this one isn't really wrapping a L0 object, but a supplementary info only. If, one day we get rid of the map, then we will do what you suggest.